components/trello/card-updates.js

```diff
@@ -0,0 +1,73 @@
+const trello = require("https://github.com/PipedreamHQ/pipedream/components/trello/trello.app.js");
+const _ = require("lodash");
+
+module.exports = {
+  name: "Card Updates",
+  description: "Emits an event for each update to a Trello card.",
+  version: "0.0.1",
+  props: {
+    trello,
+    boardId: { propDefinition: [trello, "boardId"] },
+    cardIds: {
+      propDefinition: [trello, "cardIds", (c) => ({ boardId: c.boardId })],
+    },
+    db: "$.service.db",
+    http: "$.interface.http",
+  },
+
+  hooks: {
+    async activate() {
+      let modelId = this.boardId;
+      if (!this.boardId) {
+        const member = await this.trello.getMember("me");
+        modelId = member.id;
+      }
+      const { id } = await this.trello.createHook({
+        id: modelId,
+        endpoint: this.http.endpoint,
+      });
+      this.db.set("hookId", id);
+      this.db.set("boardId", this.boardId);
+      this.db.set("cardIds", this.cardIds);
+    },
+    async deactivate() {
+      console.log(this.db.get("hookId"));
+      await this.trello.deleteHook({
+        hookId: this.db.get("hookId"),
+      });
+    },
+  },
+
+  async run(event) {
+    this.http.respond({
+      status: 200,
+    });
+
+    const body = _.get(event, "body");
+    if (body) {
+      const eventType = _.get(body, "action.type");
+      const cardId = _.get(body, "action.data.card.id");
+      const boardId = this.db.get("boardId");
+      const cardIds = this.db.get("cardIds");
+      let emitEvent = false;
+      let card;
+
+      if (eventType && eventType == "updateCard") {
```

Comment:
Does Trello have any mechanism for creating a webhook that will only notify us of specific events? i.e. can we set the webhook to only send `updateCard` events instead of receiving all events?

I think it makes sense to do the filter here either way but just curious if we can reduce the number of incoming HTTP requests we get.

Reply:
It allows you to filter by idModel (id of a board, card, list), but not by the event type that I can tell. https://developer.atlassian.com/cloud/trello/rest/api-group-webhooks/#api-webhooks-post